### ISSUE TITLE
set_node_nics cannot handle gateway change

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#298]
**Reported by bli on 7 Mar 2007 20:36 UTC**
After changing the gateway of the headnode, I got the following error when I run `install_cluster`:

```
DBD::mysql::db do failed: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1 at /opt/oscar-5.1a1r5684nightly-20070207/lib/OSCAR/oda.pm line 730.
DB_DEBUG>/opt/oscar-5.1a1r5684nightly-20070207/scripts/set_node_nics:
====> in Database::update_table SQL : UPDATE Networks SET  gateway='192.168.1.254'  WHERE n_id=
Error message: Failed to update values to Networks table in database <oscar>: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1 at /opt/oscar-5.1a1r5684nightly-20070207/lib/OSCAR/oda.pm line 736.
/opt/oscar-5.1a1r5684nightly-20070207/scripts/set_node_nics: SQL command that failed was: <UPDATE Networks SET  gateway='192.168.1.254'  WHERE n_id=  > at /opt/oscar-5.1a1r5684nightly-20070207/lib/OSCAR/oda.pm line 736.
/opt/oscar-5.1a1r5684nightly-20070207/scripts/set_node_nics: failed to write out modified network private record
DBD::mysql::db do failed: Unknown column 'gateway' in 'field list' at /opt/oscar-5.1a1r5684nightly-20070207/lib/OSCAR/oda.pm line 730.
DB_DEBUG>/opt/oscar-5.1a1r5684nightly-20070207/scripts/set_node_nics:
====> in Database::update_table SQL : UPDATE Nics SET  gateway='192.168.1.254'  WHERE id=1
Error message: Failed to update values to Nics table in database <oscar>: Unknown column 'gateway' in 'field list' at /opt/oscar-5.1a1r5684nightly-20070207/lib/OSCAR/oda.pm line 736.
/opt/oscar-5.1a1r5684nightly-20070207/scripts/set_node_nics: SQL command that failed was: <UPDATE Nics SET  gateway='192.168.1.254'  WHERE id=1  > at /opt/oscar-5.1a1r5684nightly-20070207/lib/OSCAR/oda.pm line 736.
/opt/oscar-5.1a1r5684nightly-20070207/scripts/set_node_nics: failed to modify node oscar_server nic eth0 record
Couldn't initialize the nics and networks for the oscar server at ./wizard_prep line 406
Oscar Wizard preparation script failed to complete at ./install_cluster line 230.
```

I had to manually fix this in the oscar database's `Networks` table.

**Comment by dikim33** [comment]
**Comment by dikim on 24 Mar 2007 04:53 UTC**
I have done some modifications on set_node_nics but I would like to see if this works fine with all the network entries updated.[[BR]]
Bernard, if you have any test machines to change a gateway, can you please test it with my modified set_node_nics (http://www.osl.iu.edu/~dikim/set_node_nics)?
I have test this for other network entries and it seems to work fine.

**Comment by dikim33** [comment]
**Comment by dikim on 26 Mar 2007 12:57 UTC**
(In [5758]) This commit fixes #298 which is "set_node_nics cannot handle gateway change"
